CRM/Upgrade/Incremental/php/FourFive.php

```diff
@@ -303,6 +303,57 @@ static function migrateHonoreeInfo(CRM_Queue_TaskContext $ctx) {
   }
 
   /**
+   * Upgrade function.
+   *
+   * @param string $rev
+   */
+  public function upgrade_4_5_6($rev) {
+    // Task to process sql.
+    $this->addTask(ts('Upgrade DB to 4.5.6: Fix saved searches consisting of multi-choice custom field(s)'), 'updateSavedSearch');
+
+    return TRUE;
+  }
+
+  /**
+   * Update saved search for multi-select custom fields on DB upgrade
+   *
+   * @param CRM_Queue_TaskContext $ctx
+   *
+   * @return bool TRUE for success
+   */
+  static function updateSavedSearch(CRM_Queue_TaskContext $ctx) {
+    $sql = "SELECT id, form_values FROM civicrm_saved_search";
+    $dao = CRM_Core_DAO::executeQuery($sql);
+    while ($dao->fetch()) {
+      $formValues = unserialize($dao->form_values);
+      while (list($field, $data_value) = each($formValues)) {
+        if (preg_match('/^custom_/', $field) && is_array($data_value)) {
```

Comment:
@colemanw I have a doubt regarding style as when I unserialized the saved form_values in 4.5.5 I got the data-structure in this format
    [custom_7] => Array
        (
            [0] => CiviCRM_OP_OR
            [1] => value 2
            [2] => value 1
        )

    [custom_8] => Array
        (
            [0] => CiviCRM_OP_OR
            [1] => value a
            [2] => value b
        )
Not in the old-style format mentioned in the description. 

Should I simply change the code to support the expected pseudo style mentioned in the issue ?